src/pages/NoPageFound.jsx

```diff
@@ -1,42 +1,76 @@
+// React Imports
+import { useEffect, useState } from 'react';
 // React Router Imports
 import { useNavigate } from 'react-router-dom';
 // Material UI Imports
 import Box from '@mui/material/Box';
-import Card from '@mui/material/Card';
-import CardActions from '@mui/material/CardActions';
-import CardContent from '@mui/material/CardContent';
-import CardMedia from '@mui/material/CardMedia';
 import Button from '@mui/material/Button';
+import Container from '@mui/material/Container';
+import Stack from '@mui/material/Stack';
 import Typography from '@mui/material/Typography';
 
 const NoPageFound = () => {
   const navigate = useNavigate();
+  const [photoUrl, setPhotoUrl] = useState(null);
+
```

Comment:
need state to track if its loading a URL or not otherwise it pre loads the fallback then switches to the API call if it has valid data.

Reply:
This should fix it.


```js
  const navigate = useNavigate();
  const [photoUrl, setPhotoUrl] = useState(null);
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    fetch('https://api.nasa.gov/planetary/apod?api_key=DEMO_KEY')
      .then((response) => response.json())
      .then((data) => {
        // Check if the URL is a video
        const urlIsVideo = data.media_type === 'video';
        // If it isn't a video, set the photoUrl
        if (!urlIsVideo) {
          setPhotoUrl(data.url);
        }
        setIsLoading(false);
      })
      .catch((error) => {
        console.error(error);
        setIsLoading(false);
      });
  }, []);

```

and updated background condition 

```js
background: isLoading
          ? 'rgba(0, 0, 0, 0.87)'
          : `${photoUrl ? `url(${photoUrl})` : `url(/assets/notfound.webp)`} center/cover no-repeat`
```

Reply:
done